http-server/src/main/java/io/micronaut/http/server/binding/RequestArgumentSatisfier.java

```diff
@@ -140,7 +144,7 @@ public RequestBinderRegistry getBinderRegistry() {
                     }
                 } else if (bindingResult.isPresentAndSatisfied()) {
```

Comment:
Think we also need to check here for the collection of values case ie: `@QueryParam("item") List<Integer> items` called with `item=1&item=tim`

```
                } else if (bindingResult.isPresentAndSatisfied() && (!conversionContext.hasErrors() || !httpServerConfiguration.isStrictArgumentConversion())) {
```